x-pack/plugins/security_solution/public/common/components/events_viewer/index.tsx

```diff
@@ -541,53 +547,33 @@ const StatefulEventsViewerComponent: React.FC<EventsViewerProps & PropsFromRedux
-                              defaultCellActions={defaultCellActions}
-                              hasCrudPermissions={hasCrudPermissions}
-                              filters={filters}
-                              leadingControlColumns={transformedLeadingControlColumns}
-                              pagination={{
-                                pageIndex: pageInfo.activePage,
-                                pageSize: itemsPerPage,
-                                pageSizeOptions: itemsPerPageOptions,
-                                onChangeItemsPerPage,
-                                onChangePage,
-                              }}
-                            />
-                          )}
-                          {tableView === 'eventRenderedView' && (
-                            <EventRenderedView
-                              events={nonDeletedEvents}
-                              getRowRenderer={getRowRenderer}
-                              leadingControlColumns={transformedLeadingControlColumns}
-                              pagination={{
-                                pageIndex: pageInfo.activePage,
-                                pageSize: itemsPerPage,
-                                totalItemCount: totalCountMinusDeleted,
-                                pageSizeOptions: itemsPerPageOptions,
-                                showPerPageOptions: true,
-                              }}
-                              rowRenderers={rowRenderers}
-                              scopeId={tableId}
-                              onChangePage={onChangePage}
-                              onChangeItemsPerPage={onChangeItemsPerPage}
-                              additionalControls={alertBulkActions}
-                              unitCountText={unitCountText}
-                            />
-                          )}
+                          <DataTableComponent
+                            additionalControls={alertBulkActions}
+                            unitCountText={unitCountText}
+                            browserFields={browserFields}
+                            data={nonDeletedEvents}
+                            disabledCellActions={FIELDS_WITHOUT_CELL_ACTIONS}
+                            id={tableId}
+                            loadPage={loadPage}
+                            renderCellValue={renderCellValue}
+                            rowRenderers={rowRenderers}
+                            totalItems={totalCountMinusDeleted}
+                            bulkActions={bulkActions}
+                            fieldBrowserOptions={fieldBrowserOptions}
+                            defaultCellActions={defaultCellActions}
+                            hasCrudPermissions={hasCrudPermissions}
+                            filters={filters}
+                            leadingControlColumns={transformedLeadingControlColumns}
+                            pagination={{
```

Comment:
I would memoize this object, because it creates a new reference each time this component re-renders and _may_ be causing unnecessary re-renders